backend/router.ts

```diff
@@ -0,0 +1,82 @@
+import { Router, Request, Response} from "express";
+import {DatabaseFunctions} from "./repository";
+import {createHmac} from "crypto";
+
+export class MainRouter {
+    public readonly router: Router = Router()
+
+    constructor() {
+        this.router = Router()
+        this.createRoutes()
+    }
+
+    private createRoutes() {
+        this.router
+            .get('/orders', this.getOrders)
+            .post('/search', this.searchItem)
+            .put('/items/:uuid', this.updateItem)
+            .delete('/items/:uuid', this.deleteItem)
+            .post('/items/add', this.addItem)
+            .get('/basket', this.getBasket)
+            .get('/items', this.getItems)
+            .post('/checkHash', this.adminLogging)
+            .post('/addOrder', this.addOrder)
+    }
+
+    private async getOrders(req: Request, res: Response) {
+        const answer = await DatabaseFunctions.get_orders()
+        res.json(answer)
+    }
+    private async searchItem(req: Request, res: Response) {
+        const {search, minPrice, maxPrice} = req.body
+        const answer = await DatabaseFunctions.item_finder(search, minPrice, maxPrice)
+        res.json(answer)
+    }
+    private async updateItem(req: Request, res: Response) {
+        await DatabaseFunctions.update_product(req.params.uuid, req.body)
+        res.end()
+    }
+    private async deleteItem(req: Request, res: Response) {
+        await DatabaseFunctions.remove_product(req.params.uuid)
+        res.end()
+    }
+    private async addItem(req: Request, res: Response) {
+        await DatabaseFunctions.add_new_product(req.body)
+        res.end()
+    }
+    private async getBasket(req: Request, res: Response) {
+        const result = await DatabaseFunctions.show_basket()
+        res.json(result)
+    }
+    private async getItems(req: Request, res: Response) {
+        const result = await DatabaseFunctions.select_all()
+        res.json(result)
+    }
+    private async adminLogging(req: Request, res: Response) {
+        //password ----> 'SafeBrowserShop1'

```

Comment:
Cool, I have added new product to your deployed application. But someone naughty could add, for example, racist content - and then recruiters could see that.